src/document/operation/edit_operation.ts

```diff
@@ -70,6 +70,9 @@ export class EditOperation extends Operation {
         this.maxCreatedAtMapByActor,
         this.getExecutedAt(),
       );
+      if (this.fromPos.compare(this.toPos) !== 0) {
+        root.registerRemovedNodeTextElement(text);
```

Comment:
It's strange that we don't get a compilation error even though `Text` and `RichText` don't extend `TextElement`.

Reply:
In my view, `TextElement` extend `JSONElement`, so we don't get compilation error. hmm but I'm not sure 🤔

Reply:
I think it is good to modify the code so that `RichText` and `PlainText` inherit `TextElement`. Because you can express code more strictly.

However, more than two levels of inheritance can increase the complexity of code. So I always try to use interface if there is no method to inherit.  This problem also can also be solved by changing `TextElement` to an interface.

Reply:
I thought it was strange that structural subtyping (or duck typing) works for abstract classes, not only interfaces. The weirdness I was thinking of is similar to the answer in this Stackoverflow [post](https://stackoverflow.com/questions/48829743/why-duck-typing-is-allowed-for-classes-in-typescript).